DICOM/src/Microsoft.Health.Anonymizer.Common/RedactFunction.cs

```diff
@@ -0,0 +1,155 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using EnsureThat;
+using Microsoft.Health.Anonymizer.Common.Exceptions;
+using Microsoft.Health.Anonymizer.Common.Models;
+using Microsoft.Health.Anonymizer.Common.Settings;
+
+namespace Microsoft.Health.Anonymizer.Common
+{
+    public class RedactFunction
+    {
+        private const string ReplacementDigit = "0";
+        private const int InitialDigitsCount = 3;
+        private readonly RedactSetting _redactSetting;
+
+        public RedactFunction(RedactSetting redactSetting)
+        {
+            EnsureArg.IsNotNull(redactSetting, nameof(redactSetting));
+
+            _redactSetting = redactSetting ?? new RedactSetting();
+        }
+
+        public string Redact(string inputString, AnonymizerValueTypes valueType)
+        {
+            if (inputString == null)
+            {
+                return null;
+            }
+
+            switch (valueType)
+            {
+                case AnonymizerValueTypes.Date:
+                    return RedactDateTime(inputString, DateTimeGlobalSettings.DateFormat);
+                case AnonymizerValueTypes.DateTime:
+                    return RedactDateTime(inputString, DateTimeGlobalSettings.DateTimeFormat);
+                case AnonymizerValueTypes.Age:
+                    try
+                    {
+                        return RedactAge(decimal.Parse(inputString)).ToString();
+                    }
+                    catch
+                    {
+                        throw new AnonymizerException(AnonymizerErrorCode.RedactFailed, "The input value is not a numeric age value.");
+                    }
+
+                case AnonymizerValueTypes.PostalCode:
+                    return RedactPostalCode(inputString);
+                default:
+                    return null;
+            }
+        }
+
+        public uint? Redact(uint value, AnonymizerValueTypes valueType)
+        {
+            return valueType switch
+            {
+                AnonymizerValueTypes.Age => RedactAge(value),
+                _ => null,
+            };
+        }
+
+        public decimal? Redact(decimal value, AnonymizerValueTypes valueType)
```

Comment:
Is this function used anywhere?

Reply:
Not be used now but age could be decimal.